### ISSUE TITLE
Bug: [no-unnecessary-boolean-literal-compare] Quick Fix generates bug with `instanceof`

### ISSUE THREAD

**Issue by jeengbe** [typescript-eslint/typescript-eslint#6567]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF3gDwNwFgAoIgSwDMAKbeE9DKFMEJM+AURhiRngF5-4ZKBDQgAlPADeAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ARhw5IhTWo3yJoQ+LTIcAtsRGJ0UAO4imkcGAC+IM0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "warn"
  }
}
```

### Repro Code

```typescript
declare const x;

if(x instanceof Error === false) {}
```

is turned into

```ts
declare const x;

if(!x instanceof Error) {}
```

which is semantically different:

Operator precedence: `===` < `instanceof` < `!`

### Expected Result

Firstly, there should be an option to explicitly allow `(identifier instanceof identifier === false)`.\
I prefer it over `(!(identifier instanceof identifier))`.

As for the actual bug, the following code is expected (with the aforementioned example and the above option disabled):
```ts
declare const x;

if(!(x instanceof Error)) {}
```

**Comment by bradzacher** [comment]
> if(x instanceof Error === false) {}

We'll that is certainly a very verbose and unnecessary boolean literal comparison.

---

We have utils internally to determine operator precedence so we can use those to correct the fixer.

**Comment by jeengbe** [comment]
> We'll that is certainly a very verbose and unnecessary boolean literal comparison.

While true that it is verbose, I find it by no means unnecessary given the alternative of double brackets

**Comment by jeengbe** [comment]
> > We'll that is certainly a very verbose and unnecessary boolean literal comparison.
> 
> While true that it is verbose, I find it by no means unnecessary given the alternative of double brackets

I must correct myself, please do not add that option. 

I just learned that `=== false` does not type narrow at all, so if anything, that should be its own rule: https://github.com/microsoft/TypeScript/issues/9508#issuecomment-260504948

**Comment by armano2** [comment]
while investigating this issue i noticed one more bug,
```ts
declare const x;
if ((x instanceof Error) === false) {
}
```
its going to be fixed to
```
declare const x;
if (!(x instanceof Error) {
}
```
